src/Val.cc

```diff
@@ -1878,6 +1878,10 @@ void TableVal::SetAttrs(detail::AttributesPtr a)
 		broker_store = c->AsStringVal()->AsString()->CheckString();
 		broker_mgr->AddForwardedStore(broker_store, {NewRef{}, this});
 		}
+
+	// Cache &default or &default_insert attribute.
+	if ( def_attr = attrs->Find(detail::ATTR_DEFAULT); ! def_attr )
+		def_attr = attrs->Find(detail::ATTR_DEFAULT_INSERT);
```

Comment:
General question about this: why do we need to cache the attribute given that we're already caching the value itself  in `def_val` as well? Is that only so that now we can say whether the default is of the new insert-type? If so, I'm wondering if there's another way to store that bit of information, and/or if it's worth doing the caching at all. I'm worried about making table val instances even larger than they already are; there can be many for some nested tables.

Reply:
The  commit adding it came from the opposite angle: Removing repeated `GetAttr()` lookups in favor of caching/storing it once accepting the increased memory usage. I'll rework/revert to back to repeated lookups as this is what we do elsewhere.